core/src/main/java/org/pentaho/di/core/vfs/KettleVFS.java

```diff
@@ -132,71 +131,73 @@ public static FileObject getFileObject( String vfsFilename, FileSystemOptions fs
+    }
+
     try {
-      fsOptionsForScheme = fsOptions;
       FileSystemManager fsManager = getInstance().getFileSystemManager();
       int timeOut = TIMEOUT_LIMIT;
-      boolean relativeFilename = true;
-      String[] initialSchemes = fsManager.getSchemes();
-
-      relativeFilename = checkForSchemeProvider( initialSchemes, relativeFilename, vfsFilename, space, fsOptionsForScheme );
+      String[] schemes = fsManager.getSchemes();
 
-      boolean hasScheme = vfsFilename != null && hasScheme( vfsFilename, PROVIDER_PATTERN_SCHEME );
+      String scheme = getScheme( schemes, vfsFilename );
 
       //Waiting condition - PPP-4374:
-      //We have to check for hasScheme even if it is marked as a relative path because that scheme could not
-      //be available by getSchemes at the time we validate our relativeFilename flag initially (Kitchen loading problem)
-      //So we check if - even it is marked as relative path - our vfsFilename has a possible scheme format (PROVIDER_PATTERN_SCHEME)
-      //If it does, then give it some time to be loaded. It stops when timeout is up or a scheme is found.
-
-      while ( relativeFilename && hasScheme && timeOut > 0 ) {
-        String[] schemes = fsManager.getSchemes();
-        try {
-          Thread.sleep( TIME_TO_SLEEP_STEP );
-          timeOut -= TIME_TO_SLEEP_STEP;
-          relativeFilename = checkForSchemeProvider( schemes, relativeFilename, vfsFilename, space, fsOptionsForScheme );
-        } catch ( InterruptedException e ) {
-          relativeFilename = false;
-          Thread.currentThread().interrupt();
-          break;
+      //We have to check for hasScheme even if scheme is null because that scheme could not
+      //be available by getScheme at the time we validate our scheme flag ( Kitchen loading problem )
+      //So we check if - even it has not a scheme - our vfsFilename has a possible scheme format (PROVIDER_PATTERN_SCHEME)
+      //If it does, then give it some time and tries to load. It stops when timeout is up or a scheme is found.
+
+      if ( hasSchemePattern( vfsFilename, PROVIDER_PATTERN_SCHEME ) ) {
+        while ( scheme == null && timeOut > 0 ) {
+          // ask again to refresh schemes list
+          schemes = fsManager.getSchemes();
+          try {
+            Thread.sleep( TIME_TO_SLEEP_STEP );
+            timeOut -= TIME_TO_SLEEP_STEP;
+            scheme = getScheme( schemes, vfsFilename );
+          } catch ( InterruptedException e ) {
+            Thread.currentThread().interrupt();
+            break;
+          }
         }
       }
 
-      String filename = normalizePath( vfsFilename, relativeFilename );
+      fsOptions = getFileSystemOptions( scheme, vfsFilename, space, fsOptions );
 
-      return myGetFileObject( filename );
+      String filename = normalizePath( vfsFilename, scheme );
+
+      return fsOptions != null ? fsManager.resolveFile( filename, fsOptions ) : fsManager.resolveFile( filename );
```

Comment:
We resolve the file with a normalized filename and with a fsOptions built without the normalized filename. Shouldn't fsOptions use the normalized as well?

Reply:
This was made to keep the same behavior from previous versions.